blackjack.rb

```diff
@@ -55,19 +56,15 @@ def discard
     self.betting_box = 0
   end
 
-  def deal(to_player, to_dealer=0) # deal(num_to_player, num_to_dealer)
-    while (to_player + to_dealer) > 0
-      player_hand << shoe.undelt.shift if to_player > 0
-      dealer_hand << shoe.undelt.shift if to_dealer > 0
-      to_player -= 1 if (to_player > 0)
-      to_dealer -= 1 if (to_dealer > 0)
-    end
-    "You have #{player_hand.join(" and ")} (#{player_score})...Dealer shows #{dealer_hand.first}"
+  def deal(to_player, to_dealer = 0)
+      to_player.times {player_hand << shoe.shift if to_player > 0}
+      to_dealer.times {dealer_hand << shoe.shift if to_dealer > 0}
+    "You have #{player_hand.join(' and ')} (#{player_score})...Dealer shows #{dealer_hand.first}"
   end
 
   def hit?
-    print "(h)it or (s)tay? "
-    player_score if gets == "h\n"
+    print '(h)it or (s)tay?'
+    player_score if gets.downcase.chomp == 'h'
```

Comment:
This should work, in theory...but it's sort of nice having a new-line.

And is this related to the 'h' hit issue?